engine/table/src/main/java/io/deephaven/engine/table/impl/PushdownResult.java

```diff
@@ -29,27 +36,128 @@ public class PushdownResult implements SafeCloseable {
      */
     private final WritableRowSet maybeMatch;
 
-    private PushdownResult(
-            final WritableRowSet match,
-            final WritableRowSet maybeMatch) {
-        this.match = match;
-        this.maybeMatch = maybeMatch;
+    /**
+     * Constructs a new result with an {@link RowSetFactory#empty() empty} {@link #match() match} and {@code selection}
+     * for the {@link #maybeMatch() maybe match}.
+     *
+     * <p>
+     * Equivalent to {@code ofFast(RowSetFactory.empty(), selection)}.
+     *
+     * @param selection the selection
+     * @return the result
+     */
+    public static PushdownResult maybeMatch(final WritableRowSet selection) {
+        return ofFast(RowSetFactory.empty(), selection);
+    }
+
+    /**
+     * Constructs a new result with the exact the {@code selection} as {@link #match() match} and an
+     * {@link RowSetFactory#empty() empty} {@link #maybeMatch() maybe match}.
+     *
+     * <p>
+     * Equivalent to {@code ofFast(selection, RowSetFactory.empty())}.
+     *
+     * @param selection the selection
+     * @return the result
+     */
+    public static PushdownResult match(final WritableRowSet selection) {
+        return ofFast(selection, RowSetFactory.empty());
     }
 
+    /**
+     * Constructs a new result with {@code match} and {@code maybeMatch}, which are checked to not
+     * {@link RowSet#overlaps(RowSet) overlap}. Callers that are careful in their construction may prefer to call
+     * {@link #ofFast(WritableRowSet, WritableRowSet)}.
+     *
+     * @param match rows that match
+     * @param maybeMatch rows that might match
+     * @return the result
+     */
     public static PushdownResult of(
             final WritableRowSet match,
             final WritableRowSet maybeMatch) {
+        if (match.overlaps(maybeMatch)) {
+            throw new IllegalArgumentException("match and maybeMatch should be non-overlapping row sets");
+        }
+        return ofFast(match, maybeMatch);
+    }
+
+    /**
+     * Constructs a new result with {@code match} and {@code maybeMatch}, which must not {@link RowSet#overlaps(RowSet)
+     * overlap}. This does <b>not</b> check if the row sets overlap.
+     *
+     * @param match rows that match
+     * @param maybeMatch rows that might match
+     * @return the result
+     */
+    public static PushdownResult ofFast(WritableRowSet match, WritableRowSet maybeMatch) {
```

Comment:
`ofFast()` sounds weird, IMO. what if we go with `create()` / `createUnsafe()` or some variant to match some of the other unchecked routines we use in DH caode